pkgs/by-name/pa/payme/package.nix

```diff
@@ -13,6 +13,13 @@ buildGoModule rec {
 
   vendorHash = null;
 
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.gitRefName=v${version}"
+    "-X main.gitCommit=416d53e3f518898a0411889a5af08e8d9858e70e"
```

Comment:
I'm not sure if hard coding the commit here is a great idea.
This is probably pretty easy to overlook and probably (i'm not sure on that?) can't get updated by r-ryantm

Reply:
Good catch, this won't be updated. We should just hardcode this to a dummy value

Reply:
IIRC r-ryantm runs the updateScript. Then we (?) can write a script that updates these values.

Reply:
Thanks for the feedback, I moved the commit to a variable in 8bc5eab, is there a better option?

Reply:
Is this value useful somehow? 

If it isn't, then we can do as @SuperSandro2000 suggested and fill  it with zeros or with `deadfish` five times - `deadfishdeadfishdeadfishdeadfishdeadfish`.
Else, I need to re-read the manuals on how to write updater scripts...

Reply:
I searched the nixpkgs repository for possible solutions. There seem to be two solutions to this. The first is to hard code the commit like I did in commit f4f5a30 (see [1](https://github.com/NixOS/nixpkgs/blob/f16f1016e6d1ab3253d1f83829f0c5241ecb9aaf/pkgs/by-name/cr/crc/package.nix#L15), [2](https://github.com/NixOS/nixpkgs/blob/f16f1016e6d1ab3253d1f83829f0c5241ecb9aaf/pkgs/tools/security/notary/default.nix#L6) or [3](https://github.com/NixOS/nixpkgs/blob/f16f1016e6d1ab3253d1f83829f0c5241ecb9aaf/pkgs/servers/blockbook/default.nix#L21) for previous work). Then there is the second option, which stores the commit in a temporary file to use later (see [1](https://github.com/NixOS/nixpkgs/blob/d7427e08de168ffc96cd59662934792be39390e4/pkgs/tools/security/scorecard/default.nix#L25), [2](https://github.com/NixOS/nixpkgs/blob/936ae679c9888b4c7adecc39189d4d68e72b25c0/pkgs/by-name/or/orchard/package.nix#L17) or [3](https://github.com/NixOS/nixpkgs/blob/936ae679c9888b4c7adecc39189d4d68e72b25c0/pkgs/by-name/or/orchard/package.nix#L17) for previous work).

The second option seems the most automated one. I tried using this second option ,with a temporary COMMIT file, but could not get it to work. Even though it works if I try to build one of the packages above.

```nix
  src = fetchFromGitHub {
    owner = "jovandeginste";
    repo = "payme";
    rev = "v${version}";
    hash = "sha256-WE/sAs0VSeb5UKkUy1iyjyXtgDmlQhdZkw8HMMSbQiE=";
    leaveDotGit = true;
    postFetch = ''
      git -C $out rev-parse HEAD > $out/COMMIT
      find "$out" -name .git -print0 | xargs -0 rm -rf
    '';
  };

  [...]
  
   preBuild = ''
    ldflags+=" -X main.gitCommit=$(cat COMMIT)"
  '';
```

The build fails as it can't find the COMMIT file. I don't know if it's never created (as in the postFetch is not executed) or if I am pointing to the wrong location in the preBuild step?

```
cat: COMMIT: No such file or directory
```

Any idea what I am missing here?

Reply:
@AndersonTorres As for your question, whether this is useful? It depends on your point of view, I suppose. The only thing I am trying to achieve here is adding the commit to the version string; it doesn't "change" the application, but adding the commit to the application's version output would make the nix version consistent with the binary compiled by the application developer.

Reply:
I got it; the changes in 39b733e find the commit dynamically; it's no longer hard-coded in the file.